src/components/AllWorkOrders.js

```diff
@@ -0,0 +1,193 @@
+import React, { Component } from 'react'
+import { Link } from 'react-router-dom'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faTruck, faMapMarkerAlt } from '@fortawesome/free-solid-svg-icons'
+
+import api from '../queries/api'
+import { workOrderFields } from '../queries/fields'
+import { signalsWorkOrderStatuses } from '../constants/statuses'
+
+const fields = workOrderFields.baseFields
+const statuses = signalsWorkOrderStatuses
+
+class AllWorkOrders extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            allWorkOrdersData: [],
+            location: '',
+            currentPage: 1,
+            lastPage: 1,
+        }
+    }
+
+    componentDidMount() {
+        api.allWorkOrders()
+            .getAll()
+            .then(res => {
+                this.setState({
+                    allWorkOrdersData: res.data.records,
+                    lastPage: res.data.total_pages,
+                })
+                console.log(res.data)
+            })
+    }
+
+    handleChange = event => {
+        this.setState({
+            location: event.target.value,
+        })
+    }
+
+    handleSubmit = event => {
+        event.preventDefault()
+        api.allWorkOrders()
+            .searchAll(this.state.location, 1)
+            .then(res => {
+                this.setState({
+                    allWorkOrdersData: res.data.records,
+                    lastPage: res.data.total_pages,
+                    currentPage: 1,
+                })
+                console.log(res.data)
+            })
+    }
+
+    updatePage = pageNumber => {
+        api.allWorkOrders()
+            .searchAll(this.state.location, pageNumber)
+            .then(res => {
+                this.setState({
+                    allWorkOrdersData: res.data.records,
+                    lastPage: res.data.total_pages,
+                    currentPage: pageNumber,
+                })
+                console.log(res.data)
+            })
+        window.scrollTo(0, 0)
+    }
+
+    prevPage = event => {
+        event.preventDefault()
+        // if currentPage !== 1, API call for prev page
+        console.log(typeof this.state.currentPage)
+        if (this.state.currentPage !== 1) {
+            const prevPage = this.state.currentPage - 1
+            this.updatePage(prevPage)
+        }
+        console.log('Prev')
+    }
+
+    nextPage = event => {
+        event.preventDefault()
+        // if currentPage === lastPage, nothing, else API call for next page
+        if (this.state.currentPage !== this.state.lastPage) {
+            const nextPage = this.state.currentPage + 1
+            this.updatePage(nextPage)
+        }
+        console.log('Next')
+    }
+
+    render() {
+        // make sure the data is not an empty object `{}`
+        const isMyJobsDataLoaded = this.state.allWorkOrdersData.length > 0
```

Comment:
I really like how you turned a piece of state into a boolean this way for later use in the render function. Maybe we call it `isJobsDataLoaded` instead since "My" made me think about the My Work Orders page.

Reply:
That makes sense because I used the skeleton from MyWorkOrders.js for this component. 🙃  I changed the boolean to isJobsDataLoaded since it a more suitable name for **All** Work Orders.